group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1176,4 +1176,34 @@ private <IN, OUT> OUT handleOperationException(
                 return handler.apply(apiError.error(), apiError.message());
         }
     }
+
+    /**
+     * This is the handler used by offset fetch operations to convert errors to coordinator errors.
+     *
+     * @param request           The OffsetFetchRequestGroup request.
+     * @param exception         The exception to handle.
+     * @return The OffsetFetchRequestGroup response.
+     */
+    private OffsetFetchResponseData.OffsetFetchResponseGroup handleOffsetFetchException(
+        OffsetFetchRequestData.OffsetFetchRequestGroup request,
+        Throwable exception
+    ) {
+        ApiError apiError = ApiError.fromThrowable(exception);
+
+        switch (apiError.error()) {
```

Comment:
I think that we should actually handle all the errors handled by `handleOperationException` too. A write operation could actually return some of those at any time. What do you think?

Handling `UNKNOWN_SERVER_ERROR` while we are here will also be a nice addition.

Reply:
I've updated the handler to fall through to `handleOperationException`.